### PR TITLE
fixes #8878 - ensure domain fact is downcased prior to import

### PR DESCRIPTION
**Pull request by jobewan**

### CODE REVIEW

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -415,6 +415,7 @@ def self.import_host_and_facts(hostname, facts, certname = nil, proxy_id = nil)
     # downcase everything
     hostname.try(:downcase!)
     certname.try(:downcase!)
+    facts['domain'].try(:downcase!)
```
Comment (shlomizadok) [inline app/models/host/managed.rb:418]: Maybe `facts['domain'].to_s.downcase!` ? so no need to try

Comment (jobewan) [inline app/models/host/managed.rb:418]: I thought of the same thing, however hostname and certname are utilizing try, so I was just keeping it consistent.  Let me know if the change is necessary.

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as https://github.com/theforeman/foreman/commit/25a647eb04ac2d31af99c85e447d58d1de4185e6 , thanks @jobewan

**Pull request closed without merging**